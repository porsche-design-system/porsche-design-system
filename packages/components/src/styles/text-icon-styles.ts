import type { Theme } from '@porsche-design-system/utilities-v2';
import type { TextColor } from '../types';
import type { TextColorDeprecated } from '../components/text/text-color';
import type { HeadingColor } from '../components/heading/heading-utils';
import type { HeadlineColor } from '../components/headline/headline-utils';
import type { DisplayColor } from '../components/display/display-utils';
import { getThemedColors } from './colors';

export const getThemedTypographyColor = (
  theme: Theme,
  textColor: Exclude<TextColor, TextColorDeprecated> | HeadlineColor | HeadingColor | DisplayColor
): string => {
  const {
    primaryColor,
    contrastHighColor,
    contrastMediumColor,
    contrastLowColor,
    successColor,
    errorColor,
    warningColor,
    infoColor,
  } = getThemedColors(theme);

  const colorMap: Record<
    Exclude<TextColor, TextColorDeprecated> | HeadlineColor | HeadingColor | DisplayColor,
    string
  > = {
    primary: primaryColor,
    default: primaryColor, // deprecated but part of HeadlineColor
    'contrast-low': contrastLowColor,
    'contrast-medium': contrastMediumColor,
    'contrast-high': contrastHighColor,
    'notification-success': successColor,
    'notification-warning': warningColor,
    'notification-error': errorColor,
    'notification-info': infoColor,
    inherit: 'currentColor',
  };
  return colorMap[textColor];
};
