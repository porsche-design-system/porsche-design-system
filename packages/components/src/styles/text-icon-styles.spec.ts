import { getThemedTypographyColor } from './text-icon-styles';

describe('getThemedTextColor()', () => {
  it.each<Parameters<typeof getThemedTypographyColor>>([
    ['light', 'default'], // TODO: shall be deprecated with v3-alpha
    ['light', 'contrast-high'],
    ['light', 'contrast-medium'],
    ['light', 'contrast-low'],
    ['light', 'notification-success'],
    ['light', 'notification-warning'],
    ['light', 'notification-error'],
    ['light', 'inherit'],
    ['dark', 'default'], // TODO: shall be deprecated with v3-alpha
    ['dark', 'contrast-high'],
    ['dark', 'contrast-medium'],
    ['dark', 'contrast-low'],
    ['dark', 'notification-success'],
    ['dark', 'notification-warning'],
    ['dark', 'notification-error'],
    ['dark', 'inherit'],
  ])('should return correct css for theme: %s and textColor: %s', (...args) => {
    expect(getThemedTypographyColor(...args)).toMatchSnapshot();
  });
});
