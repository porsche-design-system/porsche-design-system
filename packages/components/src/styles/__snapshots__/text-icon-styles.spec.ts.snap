// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getThemedTextColor() should return correct css for theme: dark and textColor: contrast-high 1`] = `"#AFB0B3"`;

exports[`getThemedTextColor() should return correct css for theme: dark and textColor: contrast-low 1`] = `"#404044"`;

exports[`getThemedTextColor() should return correct css for theme: dark and textColor: contrast-medium 1`] = `"#7E7F82"`;

exports[`getThemedTextColor() should return correct css for theme: dark and textColor: default 1`] = `"#FBFCFF"`;

exports[`getThemedTextColor() should return correct css for theme: dark and textColor: inherit 1`] = `"currentColor"`;

exports[`getThemedTextColor() should return correct css for theme: dark and textColor: notification-error 1`] = `"#CB3333"`;

exports[`getThemedTextColor() should return correct css for theme: dark and textColor: notification-success 1`] = `"#00C77E"`;

exports[`getThemedTextColor() should return correct css for theme: dark and textColor: notification-warning 1`] = `"#DDB84B"`;

exports[`getThemedTextColor() should return correct css for theme: light and textColor: contrast-high 1`] = `"#535457"`;

exports[`getThemedTextColor() should return correct css for theme: light and textColor: contrast-low 1`] = `"#D8D8DB"`;

exports[`getThemedTextColor() should return correct css for theme: light and textColor: contrast-medium 1`] = `"#949598"`;

exports[`getThemedTextColor() should return correct css for theme: light and textColor: default 1`] = `"#010205"`;

exports[`getThemedTextColor() should return correct css for theme: light and textColor: inherit 1`] = `"currentColor"`;

exports[`getThemedTextColor() should return correct css for theme: light and textColor: notification-error 1`] = `"#E7323B"`;

exports[`getThemedTextColor() should return correct css for theme: light and textColor: notification-success 1`] = `"#32B85B"`;

exports[`getThemedTextColor() should return correct css for theme: light and textColor: notification-warning 1`] = `"#FECC1B"`;
